packages/web-components/src/search-widget/client/components/light-box/widget.svelte

```diff
@@ -0,0 +1,180 @@
+  }
+
+  const toggleCarouselAutoPlay = () => {
+    autoPlay = !autoPlay
+  }
+
+  $: {
+    if (autoPlay) {
+      intervalId = setInterval(widgetNextButton, autoPlayInterval)
+    } else {
+      clearInterval(intervalId)
+    }
+  }
+
+  let maxCurrentDisplayImageIndexOfCurrentCarousePage
+
+  $: minCurrentDisplayImageIndexOfCurrentCarousePage = displayItemQuantity * currentCarouselPage
+  $: indexOfLastImageInTheCarousel = imageQuantity - 1
+
+  $: {
+    maxCurrentDisplayImageIndexOfCurrentCarousePage =
+      /*
+      -1 because zero based number
+    */
+      minCurrentDisplayImageIndexOfCurrentCarousePage + displayItemQuantity - 1
+
+    if (maxCurrentDisplayImageIndexOfCurrentCarousePage > imageQuantity - 1) {
+      maxCurrentDisplayImageIndexOfCurrentCarousePage = imageQuantity - 1
+    }
+  }
+
+  export let onCarouselSlideNext
+  export let onCarouselSlidePrev
+
+  const widgetPrevButton = () => {
+    const currentDisplayImageIndexAfterMovePrev = currentDisplayImageIndex - 1
+    if (currentDisplayImageIndexAfterMovePrev >= minCurrentDisplayImageIndexOfCurrentCarousePage) {
+      setDisplayImageIndex({
+        detail: currentDisplayImageIndexAfterMovePrev,
+      })
+      return
+    }
+
+    // If move to the prev image and
+    // Smaller than minCurrentDisplayImageIndexOfCurrentCarousePage (touch the left edge)
+    // -> proceed to slide the carousel backward
+    // And set currentDisplayImageIndex to the first display item (image) on the carousel
+    // For page 0, when move to the prev image -> move to the last page
+    if (currentCarouselPage === 0) {
+      setDisplayImageIndex({
+        detail: indexOfLastImageInTheCarousel,
+      })
+    } else {
+      setDisplayImageIndex({
+        detail: currentDisplayImageIndexAfterMovePrev,
+      })
+    }
+
+    onCarouselSlidePrev()
+  }
+
+  const widgetNextButton = () => {
+    const currentDisplayImageIndexAfterMoveNext = currentDisplayImageIndex + 1
+    if (currentDisplayImageIndexAfterMoveNext <= maxCurrentDisplayImageIndexOfCurrentCarousePage) {
+      setDisplayImageIndex({
+        detail: currentDisplayImageIndexAfterMoveNext,
+      })
+      return
+    }
+
+    // If move to the next image and
+    // touch the right edge
+    // -> proceed to slide the carousel backward
+    // And set currentDisplayImageIndex to the first display item (image) on the carousel
+    // -1 because zero based number
+    // For the last page, when move to the next iamge -> move to page 0
+    if (currentCarouselPage === maxCarouselPage - 1) {
+      setDisplayImageIndex({
+        detail: 0,
+      })
+    } else {
+      setDisplayImageIndex({
+        detail: currentDisplayImageIndexAfterMoveNext,
+      })
+    }
+    onCarouselSlideNext()
+  }
+</script>
+
+<style>
+  :global(div.light-box-widget-container.light-box-widget-container) {
+    background: #00000045;
+    display: flex;
+    color: white;
+
+    margin-top: -40px;
```

Comment:
Please can we use ems for spacing

Reply:
I completely agree @phmngocnghia, these components are all about keeping it lightweight and fast - nice work 👍